### PR TITLE
editor-theme3, number-pad: new Blockly

### PR DESCRIPTION
**Pull request by mxmou**
### Changes

Updates "customizable block colors" and "always show number pad" to work with the new version of Blockly.

### Tests

Tested both Scratch versions on Edge and Firefox.

### CODE REVIEW

**Comment by Samq64** [comment]
Hat blocks become flat after dynamically changing theme3's settings and dragging the block.
screenshot-2024-11-17-21-18-28

**Comment by Samq64** [comment]
~~Resolves #5886
Closes #7781~~

**Comment by Samq64** [comment]
The dial looks wrong on the Scratch website with some settings. I made sure Scratch's high contrast mode was off before enabling the addon.

The darker preset on the Scratch website:
screenshot-2024-11-18-12-54-45

And on new Blockly:
screenshot-2024-11-18-12-54-55

**Comment by mxmou** [comment]
> The dial is inconsistent between Scratch versions with some settings. I made sure Scratch's high contrast mode was off before enabling the addon.

On the current version of Scratch, the dial isn't affected by Scratch's high contrast mode and the addon doesn't change it either. This was changed in the new version and I think the way it looks with the addon is correct - the high contrast preset, for example, is identical to Scratch's high contrast mode:
image

**Review by Samq64 — APPROVED** [review]

File: addons/editor-theme3/theme3.js [hunk 1]
```diff
@@ -451,61 +645,127 @@ export default async function ({ addon, console, msg }) {
...
+      oldFieldVariableGetterInit.call(this);
+      this.textElement_.style.fill = textColor(this);
+    };
+  }
 
-  const oldFieldNoteAddOctaveButton = Blockly.FieldNote.prototype.addOctaveButton_;
-  Blockly.FieldNote.prototype.addOctaveButton_ = function (...args) {
+  let FieldNote;
+  if (Blockly.registry) FieldNote = Blockly.registry.getClass(Blockly.registry.Type.FIELD, "field_note");
+  FieldNote = Blockly.FieldNote;
```
Comment (DNin01) [inline addons/editor-theme3/theme3.js:766]: @mxmou If `FieldNote` always gets set to a predetermined value, then the if-statement above will ultimately have no effect. Was there supposed to be an `else` keyword on this line?

Comment (DNin01) [reply addons/editor-theme3/theme3.js:766]: Fixed in #8836.

**Pull request merged**